### ISSUE TITLE
Support pricing pinning 

### ISSUE THREAD

**Issue by n8maninger** [SiaFoundation/hostd#328]
### Description

Currently the UI enables hosts to set their prices based on an external currency. However, the prices are static and must be manually updated to reflect changes to the exchange rate. This change would add an optional mechanism to periodically check the exchange rate and automatically update the host's settings to reflect the new price.

- [x] Pull exchange rates into `hostd` from an external explorer
- [x] Periodically check the exchange rate and update pricing as necessary
- [x] Add API endpoints for setting the pinned prices
- [ ] Add support in the UI

**Comment by n8maninger** [comment]
#316 adds API support

**Comment by alexfreska** [comment]
@n8maninger a little hard to follow the API changes in the PR, could you update the endpoints in our API docs / Postman?
